### PR TITLE
Moves to Zipkin 2.18 module integration

### PR DESCRIPTION
**Pull request by adriancole**
Zipkin 2.18 no longer uses spring.factories to discover modules. We now
use yaml instead of AutoConfiguration. This results in smaller
distribution and startup times.

### CODE REVIEW

**Comment by adriancole** [comment]
the test docker image is failing from grpc context again.. We literally added this to our pom

```
Caused by: java.lang.ClassNotFoundException: io.grpc.Context
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
	at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:93)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	... 11 more
```

**Comment by adriancole** [comment]
docker test image problem was there before.. fixing.

**Comment by adriancole** [comment]
docker is fixed now.

**Comment by adriancole** [comment]
will merge on green, then afterwards check master build of travis, too, to keep me honest. I was able to run the test image like this..

```bash
docker build -t openzipkin/zipkin-gcp:test -f docker/Dockerfile .
```

then
```
docker run -d -p 9411:9411 -e JAVA_OPTS=--logging.level.zipkin2=DEBUG  -e STORAGE_TYPE=stackdriver   -e GOOGLE_APPLICATION_CREDENTIALS=/zipkin/.gcp/credentials.json   -e STACKDRIVER_PROJECT_ID=zipkin-demo    -e JAVA_OPTS=-Dlogging.level.zipkin2=DEBUG -v $HOME/.gcp:/zipkin/.gcp:ro   openzipkin/zipkin-gcp:test
```

File: docker/Dockerfile [hunk 1]
```diff
@@ -16,16 +16,21 @@ FROM openzipkin/zipkin-builder
 
 WORKDIR /code
 COPY . /code
-RUN mvn -B --no-transfer-progress package -DskipTests=true -pl autoconfigure/storage-stackdriver -am
+RUN mvn -B --no-transfer-progress package -DskipTests=true -pl module/storage-stackdriver -am
```
Comment (anuraaga) [inline docker/Dockerfile]: :( Thanks!

File: docker/Dockerfile [hunk 2]
```diff
@@ -16,16 +16,21 @@ FROM openzipkin/zipkin-builder
...
 
-COPY --from=0 /zipkin-gcp/ /zipkin/lib/
+COPY --from=0 /zipkin-gcp/ /zipkin/
 
-RUN echo > .xray_profile
+# Readback is currently not supported
+ENV QUERY_ENABLED false
 
-ENV MODULE_OPTS="-Dloader.path=lib/stackdriver -Dspring.profiles.active=stackdriver"
+env MODULE_OPTS="-Dloader.path=stackdriver -Dspring.profiles.active=stackdriver"
```
Comment (anuraaga) [inline docker/Dockerfile]: `env` should still be in caps like our other dockerfile commands

Comment (adriancole) [reply docker/Dockerfile]: https://github.com/openzipkin/zipkin-gcp/commit/aa53ef4be478be51c60de3bc335ad074d760a6aa thx it was copy/pasta from the other image which I'll also fix

**Pull request merged**